### DISCUSSION TITLE
O Desenvolvimento Web está se tornando desnecessariamente complicado?

### DISCUSSION THREAD

**Discussion by vitorbritto** [frontendbr/forum#244] [Discussão]
Bom dia. Este é um tópico que me chamou a atenção. Mais ainda pela forma como está exposto na ferramenta online, [Arguman](http://en.arguman.org/web-programming-is-getting-unnecessarily-complicated). Para os que não a conhecem, recomendo uma conferida.

Sendo assim, gostaria de trazer este tópico para o grupo debater a respeito.

O que penso:
- É desnecessário, quando o desenvolvimento requer algo mais sutil, objetivo ou direto.
- É complicado, quando o caminho que queremos para desenvolver é nitidamente complexo.

E vocês, o que acham sobre o tema?

http://en.arguman.org/web-programming-is-getting-unnecessarily-complicated

**Comment by woliveiras** [comment]
Caramba! Que negócio da hora!

Penso assim: 

Vejo que o desenvolvimento de Aplicações Web tem se tornado cada vez mais parecido com o Desenvolvimento de Software em geral (fui estudar Android e achei igualzinho trabalhar com JavaScript no navegador, só que melhor organizado/estruturado). Acho que essa complexidade faz parte de uma evolução dos Web Apps.

O surgimento de diversas libs é bom. Uma contribui com a outra e sempre podemos escolher a que melhor se encaixa no nosso projeto. A diversidade de ferramentas também é bom, gera certa concorrência e quem ganha com isso somos nós mesmos com a evolução dessas ferramentas.

Uma hora tudo isso vai se estabilizar e chegar num patamar mais usual. Por enquanto é só loucura e a próxima lib mais usada do mundo nascendo e morrendo todo dia! HAAHHUHAUH

**Reply by cdias55** [reply]
Boa noite, sabe me dizer como faço para interceptar sms de outros dispositivos ?

**Comment by woliveiras** [comment]
Eu ministrei uma talk sobre esse assunto de libs um tempo atrás: http://www.slideshare.net/williamoliveira542/voc-no-precisa-de-uma-sopa-de-letrinhas-para-criar-web-apps

**Comment by rubenmarcus** [comment]
segundo um cara de infra que trabalhou comigo Genexus e a solucao pra tudo

**Reply by cdias55** [reply]
Boa noite, sabe me dizer como faço para interceptar sms de outros dispositivos ?

**Comment by bruno-oliveira** [comment]
Vi isto partilhado no fb, e decidi vir dar a minha opinião.

Eu acho que o desenvolvimento de aplicações web, além de ser desnecessariamente complicado, acaba por ser terrível porque "satura" o mercado. De onde eu vim (Portugal) e onde estou agora (Europa, BENELUX), encontro entre 80-90% da oferta como sendo para desenvolvimento de aplicações web, front-end, back-end, full-stack e etc etc.

Empresas querem Msc e PhD's para desenvolverem aplicações web, quando simplesmente, basta usar uma plataforma RAD para fazer coisas funcionar (eg OutSystems, entre outras).  O desenvolvimento de apps web está "estragado" no sentido em que se passa 90% do tempo a corrigir bugs, e por essa hora, ja o negocio mudou de requisitos e etc. Não é sustentável.

**Comment by klarkc** [comment]
Acho que não, é saudável que existam diversas possibilidades de bibliotecas, frameworks e paradigmas de programação na Web. Porém naturalmente as pessoas vão adotando uma determinada postura, e se voltando a um determinado padrão, porque o mercado exige isso querendo ou não. Essa é a magia nesse ramo, o que me atrai no front-end, a possibilidade de sempre inovar e se modificar e sempre estar atrasado em relação a tecnologia, sempre vai haver algo novo e revolucionário para aprender :D

**Comment by bernardodiasc** [comment]
Comecei fazendo sites com tables. Antes da revolução do CSS, obviamente. Sei que muitos aqui também, e sabem como as coisas mudaram bastante.

As vezes noto muitos devs discutindo alicates enquanto o prego fica esperando para ser martelado. Aparentemente, nesse ramo, alguns alicates martelam pregos e alguns martelos fazem café...

Acho que naturalmente problemas complexos são solucionados com abordagens mais complicadas. O cuidado deve se ter para não ter que lidar com soluções mais complicadas do que os próprios problemas.

**Comment by xferreirax** [comment]
Deixo aqui a minha opinão de que em CSS claramente ajudou bastante e clarifou, 
Mas deixo aqui a minha opinão de que não adianta também perder imenso tempo a configurar algo, para que depois não seja encontrado no Google.

Mas claro que há casos e casos, vejam este site incrível de uma agência muito boa em Portugal  https://www.pixelinmotion.pt/
